### PR TITLE
DIS-148 Integration of Selected OT Tool

### PR DESCRIPTION
**Pull request by hhsons**
### What

Integration of Selected OT Tool for Population Types API. This change allows us to see and analyse requests to/from the service.

This PR does not address Kafka instrumentation, as this is being handled in the Kafka service.

### How to review

Run this service, and the services required to analyse OT data.
Query in your browser on `localhost:27300`
View OT through Jaeger UI and see that the call to this service was acknowledged by OT.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ func TestConfig(t *testing.T) {
 					ZebedeeURL:                 "http://localhost:8082",
 					EnablePermissionsAuth:      true,
 					DatasetAPIURL:              "http://localhost:22000",
+					OTExporterOTLPEndpoint:     "localhost:4317",
+					OTServiceName:              "dp-population-types-api",
```
Comment (charlescolbourn) [inline config/config_test.go:37]: needs batch timeout - see my message in the core team channel

File: main.go [hunk 2]
```diff
@@ -2,13 +2,15 @@ package main
...
 	"context"
+	goerrors "errors"
 	"os"
 	"os/signal"
 	"runtime/debug"
 	"syscall"
 
 	"github.com/pkg/errors"
 
+	dpotelgo "github.com/ONSdigital/dp-otel-go"
```
Comment (charlescolbourn) [inline main.go:13]: it strikes me that the dpotelgo here is probably unnecessary - I think that's what it's called anyway, so omitting it should work

File: main.go [hunk 3]
```diff
@@ -54,6 +56,22 @@ func run(ctx context.Context) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	// Set up OpenTelemetry
+	otelConfig := dpotelgo.Config{
+		OtelServiceName:          cfg.OTServiceName,
+		OtelExporterOtlpEndpoint: cfg.OTExporterOTLPEndpoint,
```
Comment (charlescolbourn) [inline main.go:62]: batch timeout

File: main.go (hunk 2 above)
Comment (hhsons) [reply main.go]: Removed alias

File: config/config_test.go (hunk 1 above)
Comment (hhsons) [reply config/config_test.go:37]: Added batch timeout, updated test, updated readme

File: main.go (hunk 3 above)
Comment (hhsons) [reply main.go:62]: added timeout to otelconfig

**Review by franmoore05 — APPROVED** [review]

**Pull request merged**